### PR TITLE
Release notes for 1.59

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
cc @rust-lang/release
r? @cuviper

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,213 @@
...
+Stabilized APIs
+---------------
+
+- [`std::thread::available_parallelism`][available_parallelism]
+- [`hash_map::Entry::insert_entry`][entry_insert_entry]
+- [`hash_map::VacantEntry::insert_entry`][vacant_insert_entry]
+- [`Result::copied`][result-copied]
+- [`Result::cloned`][result-cloned]
+- [`Option::copied`][option-copied]
+- [`Option::cloned`][option-cloned]
```
Comment (mati865) [inline RELEASES.md:37]: These two have been stabilized years ago.

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,213 @@
...
+- [`mem::MaybeUninit::as_ptr`][muninit_ptr]
+- [`mem::MaybeUninit::assume_init`][muninit_init]
+- [`mem::MaybeUninit::assume_init_ref`][muninit_init_ref]
+- [`ffi::CStr::from_bytes_with_nul_unchecked`][cstr_from_bytes]
+
+Cargo
+-----
+
+- [Stabilize the `strip` profile option][cargo/10088]
+- [Stabilize namespaced and weak dependency features.][cargo/10269]
```
Comment (mati865) [inline RELEASES.md:69]: Isn't it going to be stabilized in 1.60 instead of 1.59?

File: RELEASES.md (hunk 1 above)
Comment (Mark-Simulacrum) [reply RELEASES.md:37]: Ah, yeah, good catch.

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,210 @@
...
+- [Remove `--host` from cargo {publish,search,login}][cargo/10145]
+
+Compatibility Notes
+-------------------
+
+- [Refactor weak symbols in std::sys::unix][90846]
+  This may add new, versioned, symbols when building with a newer glibc, as the
+  standard library uses weak linkage rather than dynamically attempting to load
+  certain symbols at runtime.
+- [Deprecate crate_type and crate_name nested inside #![cfg_attr]][83744]
```
Comment (bjorn3) [inline RELEASES.md:79]:
```suggestion
- [Deprecate crate_type and crate_name nested inside `#![cfg_attr]`][83744]
```
would be slightly better I think.

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,210 @@
...
+- [Fix sparc64 ABI for aggregates with floating point members][91003]
+- [Warn when a `#[test]`-like built-in attribute macro is present multiple times.][91172]
+- [Add support for riscv64gc-unknown-freebsd][91284]
+- [Stabilize `-Z emit-future-incompat` as `--json future-incompat`][91535]
+
+Libraries
+---------
+
+- [Remove unnecessary bounds for some Hash{Map,Set} methods][91593]
+- [Remove P: Unpin bound on impl Stream for Pin][92020]
```
Comment (taiki-e) [inline RELEASES.md:26]: Stream is unstable, so I don't think it needs to be mentioned in the release notes.

```suggestion
```

File: RELEASES.md [hunk 5]
```diff
@@ -1,3 +1,203 @@
...
+    - [Remove `in_band_lifetimes` from `rustc_monomorphize`][91895]
+    - [Remove `in_band_lifetimes` from `rustc_passes`][91896]
+    - [Remove `in_band_lifetimes` from `rustc_privacy`][91925]
+    - [Remove `in_band_lifetimes` from `rustc_query_impl`][91923]
+    - [Remove `in_band_lifetimes` from `rustc_symbol_mangling`][91901]
+    - [Remove `in_band_lifetimes` from `rustc_traits`][91929]
+    - [Remove `in_band_lifetimes` from `rustc_trait_selection`][91904]
+    - [Remove `in_band_lifetimes` from `rustc_typeck`][91882]
+    - [Remove `in_band_lifetimes` from `proc_macro`][91906]
+    - [Remove `in_band_lifetimes` from `librustdoc`][92377]
```
Comment (cuviper) [inline RELEASES.md:122]: This is a lot of nearly-identical links -- can we just refer to #91867 instead?

**Comment by Mark-Simulacrum** [comment]
Resolved that nit -- r? @cuviper

**Comment by Mark-Simulacrum** [comment]
Going to go ahead and self-approve this so we have it ready to go for the release, since we'll be backporting it to beta/stable branches anyway.

@bors r+ rollup

**Pull request merged**